tool/live-monitor.js

```diff
@@ -0,0 +1,166 @@
+const Web3 = require('web3');
+const { getBridgeTransactionByTxHash } = require('../index');
+const networkParser = require('./network-parser');
+const Bridge = require('@rsksmart/rsk-precompiled-abis').bridge;
+const util = require('util');
+
+const DEFAULT_CHECK_EVERY_MILLIS = 1_000;
+
+const defaultCommandsValues = {
+    fromBlock: 'latest',
+    methods: [],
+    events: [],
+    pegout: false,
+    pegin: false,
+    network: 'testnet',
+    checkEveryMillis: DEFAULT_CHECK_EVERY_MILLIS,
+};
+
+const getParsedCommands = () => {
+    const commands = process.argv.filter(command => command.startsWith('--'))
+    .reduce((commands, command) => {
+        if(command.startsWith('--fromblock')) {
+            commands.fromBlock = command.slice(command.indexOf('=') + 1);
+        } else if(command.startsWith('--methods') || command.startsWith('--events')) { // Parsing commands that include an array
+            commands['methods'] = JSON.parse(command.slice(command.indexOf('=') + 1).replaceAll("'", '"'));
+        } else if(command.startsWith('--network')) {
+            commands.network = command.slice(command.indexOf('=') + 1);
+        } else if(command.startsWith('--pegout') || command.startsWith('--pegin')) {
+            const parsedCommand = command.replace('--', '');
+            commands[parsedCommand] = true;
+        } else if(command.startsWith('--checkeverymillis')) {
+            commands.checkEveryMillis = Number(command.slice(command.indexOf('=') + 1));
+        } else {
+            throw new Error(`Command '${command}' not recognized`);
+        }
+        return commands;
+    }, defaultCommandsValues);
+    return commands;
+};
+
+const PEGOUT_METHOD_SIGNATURES = {
+    releaseBtc: '0x',
+    updateCollections: '0x0c5a9990',
+    addSignature: '0xf10b9c59'
+};
+
+const PEGIN_METHOD_SIGNATURES = {
```

Comment:
I'm not sure what you mean here. If a user uses the options `--pegin --pegout` he will see both events related to peg-ins and peg-outs.

Reply:
What I mean is athat, as far as I understand, you have a list of peg-in and peg-out related methods to show when a user selects the option `--pegin` and?or `--pegout`. But I don't see a list of peg-in and peg-out related events, so it looks like those are not filtered. Am I missing something?

Reply:
`PEGOUT_METHOD_SIGNATURES` includes `updateCollections` method because most of the pegout related events are emitted during the `updateCollections` method call.

We already talked about the passed `events` option in the command. Specific emitted events depend on specific methods calls. Thus, when the `events` option is passed and the `methods` option is not passed or the `--pegin` or `--pegout` options are passed, we try to find one of those `events` in the transaction receipt `events` structure and if we find at least one event, we print all the events that the method emitted, to have a greater context.

Reply:
Ok ok I see. Thanks!